connectivity/drivers/lora/TARGET_STM32WL/STM32WL_LoRaRadio.cpp

```diff
@@ -0,0 +1,1561 @@
+
+/**
+ / _____)             _              | |
+( (____  _____ ____ _| |_ _____  ____| |__
+ \____ \| ___ |    (_   _) ___ |/ ___)  _ \
+ _____) ) ____| | | || |_| ____( (___| | | |
+(______/|_____)_|_|_| \__)_____)\____)_| |_|
+    (C)2015 Semtech
+ ___ _____ _   ___ _  _____ ___  ___  ___ ___
+/ __|_   _/_\ / __| |/ / __/ _ \| _ \/ __| __|
+\__ \ | |/ _ \ (__| ' <| _| (_) |   / (__| _|
+|___/ |_/_/ \_\___|_|\_\_| \___/|_|_\\___|___|
+embedded.connectivity.solutions===============
+
+Description: LoRaWAN stack layer that controls both MAC and PHY underneath
+
+License: Revised BSD License, see LICENSE.TXT file include in the project
+
+Maintainer: Miguel Luis, Gregory Cristian & Gilbert Menth
+
+Copyright (c) 2019, Arm Limited and affiliates.
+
+SPDX-License-Identifier: BSD-3-Clause
+*/
+/**
+  ******************************************************************************
+  *
+  *          Portions COPYRIGHT 2020 STMicroelectronics
+  *
+  * @file    STM32WL_loRaRadio.c
+  * @author  MCD Application Team
+  * @brief   radio driver implementation
+  ******************************************************************************
+  */
+
+
+#include <math.h>
+#include "ThisThread.h"
+#include "mbed_wait_api.h"
+#include "Timer.h"
+#include "STM32WL_LoRaRadio.h"
+#include "radio_board_if.h"
+#include "sys_debug.h"
+#include "stm32wlxx_hal_subghz.h"
+
+
+/**
+  * @brief Set RX pin to high or low level
+  */
+#define DBG_GPIO_RADIO_RX(set_rst) DBG_GPIO_##set_rst##_LINE(DGB_LINE1_PORT, DGB_LINE1_PIN);
+
+/**
+  * @brief Set TX pin to high or low level
+  */
+#define DBG_GPIO_RADIO_TX(set_rst) DBG_GPIO_##set_rst##_LINE(DGB_LINE2_PORT, DGB_LINE2_PIN);
+
+#if MBED_CONF_STM32WL_LORA_DRIVER_REGULATOR_MODE == 1
+	uint8_t regulator_mode = USE_DCDC;
+#else
+	uint8_t regulator_mode = USE_LDO;
+#endif
+#if MBED_CONF_STM32WL_LORA_DRIVER_CRYSTAL_SELECT== 1
+  uint8_t _crystal_select  = 1;
+#else
+	uint8_t _crystal_select = 0;
+#endif
+
+static void SUBGHZ_Radio_IRQHandler(void);
+// Handler called by thread in response to signal directly
+static void RadioIrqProcess();
+
+// Structure containing function pointers to the stack callbacks
+static radio_events_t *_radio_events;
+
+//SUBGHZ handle Structure definition
+SUBGHZ_HandleTypeDef hsubghz;
+ 
+// Data buffer used for both TX and RX
+// Size of this buffer is configurable via Mbed config system
+// Default is 255 bytes
+static uint8_t _data_buffer[MAX_DATA_BUFFER_SIZE_STM32WL];
+
+static uint8_t _operation_mode;
+static uint8_t _active_modem;
+    
+using namespace mbed;
+using namespace rtos;
```

Comment:
Shouldn't this be below line 89 ? if MBED_CONF_RTOS_PRESENT is not present I don't think that namespace rtos would exist ? 
actually do we plan to support baremetal as well ?

Reply:
Also if both rtos based and bare metal profiles are supported, it would be great to briefly explain advantage of the rtos and the differences in behavior.

Reply:
using namespace rtos; set under #define MBED_CONF_RTOS_PRESENT: done
Concerning Baremetal, this is a good question to be discussed together. I haven't looked at it yet